lib/happo/uploader.rb

```diff
@@ -18,6 +19,9 @@ def upload_diffs
 
       bucket = find_or_build_bucket
       dir = SecureRandom.uuid
+      if !(@s3_bucket_path.nil? || @s3_bucket_path.empty?)
+        dir = File.join(@s3_bucket_path, SecureRandom.uuid)
```

Comment:
Sure, or you could use the code I posted in https://github.com/Galooshi/happo/pull/92#discussion_r71615131 :D